splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/ProjectRestrictOperation.java

```diff
@@ -172,7 +172,7 @@ public void init(SpliceOperationContext context) throws StandardException, IOExc
         cloneMap = ((boolean[]) statement.getSavedObject(cloneMapItem));
 
         if (restrictionMethodName != null)
-            restriction = new SpliceMethod<>(restrictionMethodName, activation);
+            restriction = new SpliceMethod<>(restrictionMethodName, source.getActivation());
```

Comment:
It feels like we need some comment here. Why the original code is wrong? I mean, for the generated code of one query, could this PRN and its source have different activations?

Reply:
It is a theoretical bug.  If the activations ever differ, it will be broken.  Using the source activation here ensures the correct one is always used.
Added the following comment:
>         // setCurrentRow, called by the source operation, is always done in the activation
>         // used by the source.  In case this activation differs from the activation
>         // of this ProjectRestrict, use the source's activation as the activation
>         // for looking up the row in the restriction method.
>         if (restrictionMethodName != null)
>             restriction = new SpliceMethod<>(restrictionMethodName, source.getActivation());